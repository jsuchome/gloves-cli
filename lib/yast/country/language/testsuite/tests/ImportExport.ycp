/**
 * tests for Language.ycp autoyast functions
 * Author:	jsuchome@suse.cz
 * $Id: ImportExport.ycp 32769 2006-09-08 09:29:15Z jsuchome $
 */
{
    // testedfiles: Language.ycp

    import "Testsuite";

    map READ = $[
	"sysconfig" : $[
	    "language" : $[
		"DEFAULT_LANGUAGE"	: "en_US",
	    ],
	],
	"target" : $[
	    "dir" : [
		"language_de_DE.ycp"
	    ],
	    "yast2" : $[
		"de_DE" : [
		    "Deutsch",
		    "Deutsch",
		    ".UTF-8",
		    "@euro",
		    "German"
		],
	    ],
	],
    ];

    map E	= $[
	"target" : $[
	    "bash_output"	: $[],
	],
    ];

    import "Mode";
    Mode::SetMode ("autoinst_config");

    Testsuite::Init ([READ,$[],E], nil);

    import "Language";

    Testsuite::Test (``(Language::Export ()), [], nil);

    Testsuite::Test (``(Language::Modified ()), [], nil);

    map imported	= $[
	"language"	: "de_DE"
    ];

    Testsuite::Test (``(Language::Import (imported)), [READ,$[],E], nil);

    Testsuite::Test (``(Language::Modified ()), [], nil);

    Testsuite::Test (``(Language::Export ()), [], nil);

    Testsuite::Test (``(Language::Summary ()), [], nil);

    // now let's add another language
    imported	= $[
	"languages"	: "cs_CZ"
    ];

    Testsuite::Test (``(Language::Import (imported)), [$[],$[],E], nil);

    Testsuite::Test (``(Language::Export ()), [], nil);

    // cs_CZ not shown in summary, as it was not part of READ map
    Testsuite::Test (``(Language::Summary ()), [], nil);
}
