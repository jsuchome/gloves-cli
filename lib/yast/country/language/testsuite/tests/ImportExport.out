Read	.target.dir "/usr/share/YaST2/data/languages" ["language_de_DE.ycp"]
Read	.target.yast2 "languages/language_de_DE.ycp" $["de_DE":["Deutsch", "Deutsch", ".UTF-8", "@euro", "German"]]
Return	$["language":"en_US", "languages":""]
Return	false
Execute	.target.bash_output "/usr/bin/locale -a" $[]
Return	true
Return	true
Return	$["language":"de_DE", "languages":"de_DE"]
Return	<ul><li>Primary Language: German</li></ul>
Return	true
Return	$["language":"de_DE", "languages":"cs_CZ,de_DE"]
Return	<ul><li>Primary Language: German</li></ul>
